### ISSUE TITLE
Bug: Duplicate `Globals` sections in `hello-world-powertools-dotnet` 

### ISSUE THREAD

**Issue by ljacobsson** [aws/aws-sam-cli#4860]
### Description:
If I select 'y' on both xray and applications insights prompts, then the tool generates two `Globals` sections:

```
$ sam init --app-template hello-world-powertools-dotnet --name sam-app --package-type Zip --runtime dotnet6

Based on your selections, the only Template available is Hello World Example.
We will proceed to selecting the Template as Hello World Example.

Based on your selections, the only dependency manager available is cli-package.
We will proceed copying the template using cli-package.

Would you like to enable X-Ray tracing on the function(s) in your application?  [y/N]: y
X-Ray will incur an additional cost. View https://aws.amazon.com/xray/pricing/ for more details

Would you like to enable monitoring using CloudWatch Application Insights?
For more info, please view https://docs.aws.amazon.com/AmazonCloudWatch/latest/monitoring/cloudwatch-application-insights.html [y/N]: y
AppInsights monitoring may incur additional cost. View https://docs.aws.amazon.com/AmazonCloudWatch/latest/monitoring/appinsights-what-is.html#appinsights-pricing for more details

Cloning from https://github.com/aws/aws-sam-cli-app-templates (process may take a moment)

    -----------------------
    Generating application:
    -----------------------
    Name: sam-app
    Runtime: dotnet6
    Architectures: x86_64
    Dependency Manager: cli-package
    Application Template: hello-world-powertools-dotnet
    Output Directory: .
    Configuration file: sam-app/samconfig.toml
    
    Next steps can be found in the README file at sam-app/README.md
        

Commands you can use next
=========================
[*] Create pipeline: cd sam-app && sam pipeline init --bootstrap
[*] Validate SAM template: cd sam-app && sam validate
[*] Test Function in the Cloud: cd sam-app && sam sync --stack-name {stack-name} --watch


Error: while constructing a mapping
  in "./sam-app/template.yaml", line 1, column 1
found duplicate key "Globals" with value "ordereddict([('Function', ordereddict([('Tracing', 'Active')])), ('Api', ordereddict([('TracingEnabled', True)]))])" (original value: "ordereddict([('Function', ordereddict([('Timeout', 10), ('MemorySize', 256), ('Runtime', 'dotnet6')])), ('Api', ordereddict([('TracingEnabled', True)]))])")
  in "./sam-app/template.yaml", line 16, column 1

To suppress this check see:
    http://yaml.readthedocs.io/en/latest/api.html#duplicate-keys

Traceback:
  File "click/core.py", line 1055, in main
  File "click/core.py", line 1657, in invoke
  File "click/core.py", line 1404, in invoke
  File "click/core.py", line 760, in invoke
  File "samcli/commands/init/command.py", line 132, in wrapped
  File "click/decorators.py", line 84, in new_func
  File "click/core.py", line 760, in invoke
  File "samcli/lib/telemetry/metric.py", line 183, in wrapped
  File "samcli/lib/telemetry/metric.py", line 148, in wrapped
  File "samcli/lib/utils/version_checker.py", line 42, in wrapped
  File "samcli/cli/main.py", line 92, in wrapper
  File "samcli/commands/init/command.py", line 292, in cli
  File "samcli/commands/init/command.py", line 383, in do_cli
  File "samcli/commands/init/interactive_init_flow.py", line 82, in do_interactive
  File "samcli/commands/init/interactive_init_flow.py", line 149, in generate_application
  File "samcli/commands/init/interactive_init_flow.py", line 294, in _generate_from_use_case
  File "samcli/commands/init/init_generator.py", line 33, in do_generate
  File "samcli/lib/init/__init__.py", line 133, in generate_project
  File "samcli/lib/init/__init__.py", line 148, in _enable_application_insights
  File "samcli/lib/init/template_modifiers/application_insights_template_modifier.py", line 40, in __init__
  File "samcli/lib/init/template_modifiers/cli_template_modifier.py", line 19, in __init__
  File "samcli/lib/init/template_modifiers/application_insights_template_modifier.py", line 44, in _get_template
  File "ruamel/yaml/main.py", line 434, in load
  File "ruamel/yaml/constructor.py", line 121, in get_single_data
  File "ruamel/yaml/constructor.py", line 131, in construct_document
  File "ruamel/yaml/constructor.py", line 1569, in construct_yaml_map
  File "ruamel/yaml/constructor.py", line 1453, in construct_mapping
  File "ruamel/yaml/constructor.py", line 294, in check_mapping_key

An unexpected error was encountered while executing "sam init".
```

### Additional environment details (Ex: Windows, Mac, Amazon Linux etc)

1. OS: Linux - Ubuntu
2. If using SAM CLI, `sam --version`: 1.75.0
3. AWS region: anywhere

**Comment by mndeveci** [comment]
Moved this to SAM CLI repo as this is happening during post-processing of the template.

**Comment by sleepwithcoffee** [comment]
hi @mndeveci 

I think the bug is under `/workspaces/iac/aws-sam/aws-sam-cli/samcli/lib/init/template_modifiers/xray_tracing_template_modifier.py`

since the constant is `GLOBALS = "Globals:\n"`
the template has this line (no new line) `Globals: # More info about Globals: https://github.com/awslabs/serverless-application-model/blob/master/docs/globals.rst`

I think it is a flaw that this process requires directive ending with "\n"

A quick way to fix this is to update SAM template.
A better way is to update the class `XRayTracingTemplateModifier` to parse the directives regardless of the ending character.

**Comment by mndeveci** [comment]
Thanks for your investigation @sleepwithcoffee! We would be happy to review the PR once you raised it!

**Comment by hawflau** [comment]
@sleepwithcoffee as discuss in another issue (https://github.com/aws/aws-sam-cli-app-templates/pull/359), I think `XRayTracingTemplateModifier` should be smarter in modifying the template and avoiding breaking anything.

For your context, `XRayTracingTemplateModifier` is using a text-based approach to do the modification, which IMO is quite brittle and has a lot of edge cases to handle. A better approach is to use `ruamel` (which `ApplicationInsightsTemplateModifier` uses). Basically it reads the yaml file into a dict, so we can easily check if there's already an entry for a certain property.

**Comment by sleepwithcoffee** [comment]
thanks @hawflau for the pointer. I notice that this issue is currently being worked on, so I would not work on this

**Comment by qingchm** [comment]
Hey guys this has been addressed in our most recent release 1.79.0 (https://github.com/aws/aws-sam-cli/releases/tag/v1.79.0). Closing the issue here. Thanks for all the engagement!
